src/coreclr/nativeaot/System.Private.CoreLib/src/System/Runtime/InteropServices/TrackerObjectManager.NativeAot.cs

```diff
@@ -0,0 +1,543 @@
+        {
+            GC.WaitForPendingFinalizers();
+        }
+
+        public static void NotifyEndOfReferenceTrackingOnThread()
+        {
+            ComWrappers.ReleaseExternalObjectsFromCurrentThread();
+        }
+
+        // Creates a proxy object (managed object wrapper) that points to the given IUnknown.
+        // The proxy represents the following:
+        //   1. Has a managed reference pointing to the external object
+        //      and therefore forms a cycle that can be resolved by GC.
+        //   2. Forwards data binding requests.
+        //
+        // For example:
+        //
+        // Grid <---- NoCW             Grid <-------- NoCW
+        // | ^                         |              ^
+        // | |             Becomes     |              |
+        // v |                         v              |
+        // Rectangle                  Rectangle ----->Proxy
+        //
+        // Arguments
+        //   obj        - An IUnknown* where a NoCW points to (Grid, in this case)
+        //                    Notes:
+        //                    1. We can either create a new NoCW or get back an old one from the cache.
+        //                    2. This obj could be a regular tracker runtime object for data binding.
+        //  ppNewReference  - The IReferenceTrackerTarget* for the proxy created
+        //                    The tracker runtime will call IReferenceTrackerTarget to establish a reference.
+        //
+        public static void GetTrackerTarget(IntPtr punk, out IntPtr ppNewReference)
+        {
+            if (punk == IntPtr.Zero)
+            {
+                throw new ArgumentException();
+            }
+
+            if (Marshal.QueryInterface(punk, in IID_IUnknown, out IntPtr ppv) != 0)
+            {
+                throw new InvalidCastException();
+            }
+
+            using ComHolder identity = new ComHolder(ppv);
+            using ComHolder trackerTarget = new ComHolder(GetOrCreateTrackerTarget(identity.Ptr));
+            if (Marshal.QueryInterface(trackerTarget.Ptr, in IID_IReferenceTrackerTarget, out ppNewReference) != 0)
+            {
+                throw new InvalidCastException();
+            }
+        }
+
+        public static void AddMemoryPressure(long bytesAllocated)
+        {
+            GC.AddMemoryPressure(bytesAllocated);
+        }
+
+        public static void RemoveMemoryPressure(long bytesAllocated)
+        {
+            GC.RemoveMemoryPressure(bytesAllocated);
+        }
+    }
+
+    internal static class TrackerObjectManager
+    {
+        internal static readonly IntPtr s_findReferencesTargetCallback = FindReferenceTargetsCallback.CreateFindReferenceTargetsCallback();
+
+        internal static volatile IntPtr s_TrackerManager;
+        internal static volatile bool s_HasTrackingStarted;
+        internal static volatile bool s_IsGlobalPeggingOn = true;
+
+        internal static DependentHandleList s_ReferenceCache;
```

Comment:
```suggestion
        internal static volatile IntPtr s_trackerManager;
        internal static volatile bool s_hasTrackingStarted;
        internal static volatile bool s_isGlobalPeggingOn = true;

        internal static DependentHandleList s_referenceCache;
```
Nit: Repo coding conventions